commands/audit/scarunner.go

```diff
@@ -429,3 +420,30 @@ func dumpScanResponseToFileIfNeeded(results []services.ScanResponse, scanResults
 	}
 	return utils.DumpContentToFile(fileContent, scanResultsOutputDir, scanType.String())
 }
+
+func getDiffDependencyTree(targetBranchResult *results.TargetResults, sourceBranchResults *results.SecurityCommandResults, treeResult *DependencyTreeResult) (*DependencyTreeResult, error) {
+	targetSbom := targetBranchResult.Sbom
+	var sourceSbom results.Sbom
+	if sbr := sourceBranchResults.GetTargetResults(targetBranchResult.Target); sbr != nil {
+		sourceSbom = sbr.Sbom
+	} else {
+		return nil, fmt.Errorf("failed to get source branch results for target: %s", targetBranchResult.Target)
+	}
+	targetDepsMap := datastructures.MakeSet[string]()
+	for _, component := range targetSbom.Components {
+		targetDepsMap.Add(component.ToXrayComponentId())
+	}
+	addedDepsMap := datastructures.MakeSet[string]()
+	for _, component := range sourceSbom.Components {
+		componentId := component.ToXrayComponentId()
+		if exists := targetDepsMap.Exists(componentId); !exists {
+			addedDepsMap.Add(componentId)
```

Comment:
```suggestion
            // Dependency in source but not in target, relevant for diff
			addedDepsMap.Add(componentId)
```
I would add a comment here